app/controllers/users/roles_controller.rb

```diff
@@ -0,0 +1,43 @@
+class Users::RolesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :load_user
+  before_action :authorize_user
+  before_action :redirect_to_account_page_if_acting_on_own_user, only: %i[edit]
+
+  def edit; end
+
+  def update
+    updater = UserUpdate.new(@user, user_params, current_user, user_ip_address)
+    if updater.call
+      redirect_to edit_user_path(@user), notice: "Updated user #{@user.email} successfully"
+    else
+      render :edit
+    end
+  end
+
+private
+
+  def load_user
+    @user = User.find(params[:user_id])
+  end
+
+  def authorize_user
+    authorize(@user)
+    authorize(@user, :assign_role?)
+  end
+
+  def user_params
+    UserParameterSanitiser.new(
+      user_params: permitted_user_params,
+      current_user_role: current_user.role.to_sym,
+    ).sanitise
+  end
+
+  def permitted_user_params
+    @permitted_user_params ||= params.require(:user).permit(:role).to_h
+  end
+
+  def redirect_to_account_page_if_acting_on_own_user
+    redirect_to account_path if current_user == @user
```

Comment:
Given that a superadmin user can edit their own role, I wonder whether it'd be better to redirect to `edit_account_role_path` here? Non superadmins will have already been rejected by `authorize_user` so we don't need to worry about them.

Reply:
That makes sense - I thought we'd changed the equivalent page for editing user email to redirect to the account page rather than the account edit email page, but I was mistaken. I will change this to match as you suggest.

Reply:
Addressed in https://github.com/alphagov/signon/pull/2537/commits/a0e540a4a064eaff66e206892822649c278020a1